mining/worker.go

```diff
@@ -72,6 +72,10 @@ type MessageApplier interface {
 	ApplyMessagesAndPayRewards(ctx context.Context, st state.Tree, vms vm.StorageMap, messages []*types.SignedMessage, minerOwnerAddr address.Address, bh *types.BlockHeight, ancestors []types.TipSet) (consensus.ApplyMessagesResponse, error)
 }
 
+type workerPorcelainAPI interface {
```

Comment:
This _could_ be extended to replace other fields on the `DefaultWorker`, however that may be outside the scope of the PR.

Reply:
I agree it could.  Furthermore I think the mining worker should eventually be code in the block miner protocol's `miner.go` and most of the dependencies it shares with the node should be accessed through porcelain.

I also agree all of this is out of scope of your PR.